R/find-endpoint.R

```diff
@@ -0,0 +1,146 @@
+library(sf)
+library(ggplot2)
+library(dplyr)
+library(igraph)
+# group isolated parts of the road network
+road_network = st_read("outputdata/orcp_city_boundary.geojson")
+
+dist_matrix <- st_distance(road_network)
+threshold <- set_units(50, "m")
+
+connectivity_matrix <- as(dist_matrix < threshold, "lgCMatrix")
+# Identify connected components using the igraph package
+graph <- graph_from_adjacency_matrix(connectivity_matrix, mode = "undirected", diag = FALSE)
+components <- components(graph)
+road_network$component <- components$membership
+unique(road_network$component)
+map_view <- mapview::mapview(road_network, zcol = "component", legend = TRUE)
+
+# Print the map
+print(map_view)
+
+
+component_8 <- road_network[road_network$component == 8, ] 
+sf::st_write(component_8, "outputdata/component_8.geojson", delete_dsn = TRUE)
+
+# Read the GeoJSON file
+gdf <- st_read("outputdata/orcp_city_boundary.geojson") 
+
+# filter gdf based on compoent = 9
+gdf <- gdf[gdf$component == "9", ] |> 
+  st_transform(crs = 27700)
+
+plot(gdf)
```

Comment:
Maybe move code like that to a script in the `code` folder? I think this is causing `tar_source()` to fail. I get:

```
> library(tidyverse)
> library(targets)
> library(tidygraph)
> library(osmextract)
> tar_source()
Error in eval(expr = expr, envir = envir) : object 'gdf' not found
```